src/net.cpp

```diff
@@ -1344,15 +1344,47 @@ void CConnman::DisconnectNodes()
 
 void CConnman::NotifyNumConnectionsChanged()
 {
-    size_t nodes_size;
+    decltype(m_nodes) nodes_copy;
     {
         LOCK(m_nodes_mutex);
-        nodes_size = m_nodes.size();
+        nodes_copy = m_nodes;
     }
-    if(nodes_size != nPrevNodeCount) {
-        nPrevNodeCount = nodes_size;
+
+    int num_outbound_full_relay{0};
+    int num_block_relay{0};
+    int num_manual{0};
+    int num_inbound{0};
+    for (const auto* node : nodes_copy) {
```

Comment:
36e20da8867a73bdae51e211b101ffc0cbf4b55f

Correct me if I'm wrong but I think it's safe to use `node` without `AddRef`/`Release` because `DeleteNode` isn't called concurrently with `CConnman::NotifyNumConnectionsChanged`.

Reply:
Do you mind elaborating? What change do you suggest?

Reply:
That's my point, I think the correct code is fine. Initially, I wondered if some `AddRef`/`Release` was needed.